itests/f3_test.go

```diff
@@ -419,3 +427,158 @@ func setupWithStaticManifest(t *testing.T, manif *manifest.Manifest, testBootstr
 
 	return e
 }
+
+// TestAPIChainGetFinalizedTipSet tests the behavior of ChainGetFinalizedTipSet
+// under various F3 states, including F3 disabled, nominal operation,
+// F3 not ready, F3 internal errors, and implausible F3 certificates.
+// This is modelled off similar tests in api_v2_test.go for v2/ChainGetTipSet.
```

Comment:
What about the scenario where where F3 finalized tipset is more than 900 epochs behind head?

Reply:
covered in the `f3 is older than ec falls back to ec` variant in here, we finalize at 910 and expect 900